source/core/Array.cpp

```diff
@@ -1119,34 +1119,63 @@ VIREO_FUNCTION_SIGNATURET(ArrayMaxMinInternal, FindArrayMaxMinInstruction)
     IntIndex maxIndex = 0, minIndex = 0;
     AQBlock1* minValue = arrayIn->BeginAt(0);
     AQBlock1* maxValue = minValue;
-    if (len == 0)
+    if (len == 0) {
         maxIndex = minIndex = -1;
+    } else {
+        if (arrayIn->ElementType()->IsFloat()) {
+            maxIndex = minIndex = -1;
+            if (arrayIn->ElementType()->TopAQSize() == sizeof(Double)) {
+                for (IntIndex i = 0; i < len; ++i) {
+                    minValue = arrayIn->BeginAt(i);
+                    if (!::isnan(*(Double*)minValue)) {
+                        maxIndex = minIndex = i;
+                        maxValue = minValue;
+                        break;
+                    }
+                }
+            } else {
+                for (IntIndex i = 0; i < len; ++i) {
+                    minValue = arrayIn->BeginAt(i);
+                    if (!::isnan(*(Single*)minValue)) {
+                        maxIndex = minIndex = i;
+                        maxValue = minValue;
+                        break;
+                    }
+                }
+            }
+        }
+    }
+
     for (IntIndex i = 0; i < len; i++) {
-        Boolean less;
-        snippet->_p0 = arrayIn->BeginAt(i);
+        Boolean shouldUpdateMaxOrMin = false;
+        AQBlock1* currentElement = arrayIn->BeginAt(i);
+        snippet->_p0 = currentElement;
         snippet->_p1 = minValue;
-        snippet->_p2 = &less;
+        snippet->_p2 = &shouldUpdateMaxOrMin;
```

Comment:
I would keep it "less" and not rename it to "shouldUpdateMaxOrMin" as it is the "less" operator. "less" does the right thing when an input is "NaN" also (The less primitive in G also does the right thing when an input is NaN).